docs_src/design/adr/security/0009-Secure-Bootstrapping.md

```diff
@@ -49,303 +26,122 @@
+* Other one-time initialization activities.
 
-In summary, the bootstrapping secrets are:
+Workarounds when an installation phase is not present include:
 
-* Secrets seed (supplied externally to the framework in some hardened configurations)
-* Consul Serf encryption key (used to control cluster membership)
-* Consul management token (used by Vault Consul secrets engine)
-* Vault master key (used to encrypt Vault's secrets)
-* PostgreSQL superuser password
-* Redis database password
-* Kong admin credential
+* Perform initialization tasks manually,
+  and manually seed secrets into static configuration files.
+* Ship with known hard-coded secrets in static configuration files.
+* Start in an insecure configuration and remain that way.
 
-See
-[ADR 0008 - Creation and Distribution of Secrets](https://github.com/edgexfoundry/edgex-docs/blob/master/docs_src/design/adr/security/0008-Secret-Creation-and-Distribution.md)
-for a more detailed specification about how secrets are created and distributed within the framework.
+EdgeX does not have a manual installation flow,
+and uses a combination of the last two approaches.
 
-### Secrets that can't be bootstrapped
+The objective of this ADR is to define a framework for Docker-based initialization logic in EdgeX.
+This will enable the removal of certain hard-coded secrets in EdgeX
+and enable certain components (such as Consul) to be started in a secure configuration.
+These improvement are necessary pre-requisites to implementing microservice communication security.
 
-#### Vault access tokens
+### History
 
-Each EdgeX microservice will have its Vault access token provided to it
-via a filesystem-based secret.
-This token is used to bootstrap security for each microservice.
-Each Vault access token has a TTL
-and it is not possible to renew an expired token.
-New ones must be generated whenever the framework is started.
+In previous releases, container startup sequencing has been
+primarily been driven by Consul service health checks
+backed healthcheck endpoints of particular services
+or by sentinel files placed in the file system
+when certain intialization milestones are reached.
 
-Though Vault access tokens cannot be bootstrapped at install time,
-it is nevertheless important to hold off starting
-the majority of EdgeX services
-until Vault is up and access tokens generated.
+The implementation has been plagued by several issues:
 
-#### Consul access tokens
+* Sentinel files are not cleaned up if the framework fails or is shut down.
+  Invalid state left over from previous instantiations of the framework
+  causes difficult-to-resolve race conditions.
 
-Consul access tokens can be requested
-from the Vault Consul secrets engine at runtime
-on an as-needed basis.
+* Consul healh checks are reported in a difficult-to-parse
+  JSON structure, which has lead to the creation of specialized
+  tools that are insensitive to libc implementations
```

Comment:
is libc a typo?

Reply:
@cloudxxx8 No, "libc" is the C language runtime.  Programs compiled with dynamic linking enabled end up being dependent on the system's installed libc, which makes it very difficult to take an executable compiled on one Linux distro and portable run it on another Linux distro.  There are two major versions of libc: GLIBC and MUSL which are definitely not runtime compatible with each other

Reply:
thanks for the explanation